### PR TITLE
add optionselected event and confidence calculator

### PR DESCRIPTION
**Pull request by sarashunter**

### CODE REVIEW

File: server/utils/calculateConfidence.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const calculateConfidence = (timeStarted: Date): ConfidenceValue => {
+	const currentTime = new Date();
+	const millisecondsElapsed = currentTime.getTime() - timeStarted.getTime();
+	const secondsElapsed = millisecondsElapsed / 1000;
+
+	if (secondsElapsed < 15) {
+		return ConfidenceValue.high;
+	}
+	if (secondsElapsed < 45) {
```
Comment (sarashunter) [inline server/utils/calculateConfidence.ts:11]: I arbitrarily chose these numbers to start.  If we switch to having a countdown timer, then we can switch this up.

File: server/models/room.model.ts [hunk 2]
```diff
@@ -5,10 +5,16 @@ export interface User {
 	name: string;
 }
 
+export enum ConfidenceValue {
+	low = 0,
+	medium = 1,
+	high = 2,
+}
```
Comment (sarashunter) [inline server/models/room.model.ts:12]: using an enum for this made sense to me, but I'm open to switching it back.

**Comment by sarashunter** [comment]
image
Tested using postman

File: server/utils/calculateConfidence.ts (hunk 1 above)
Comment (lvl-mattwells) [reply server/utils/calculateConfidence.ts:11]: Hmm, 15 seconds seems a little slow for high confidence. What about like 5 or even 3 seconds? Then maybe medium is 15 seconds or less?

Comment (sarashunter) [reply server/utils/calculateConfidence.ts:11]: Sure.  I'll switch it to 5 and 15 for now.

File: server/controllers/socket.controller.ts [hunk 3]
```diff
@@ -5,13 +5,31 @@ import {
...
 
 const sockets = new Map<string, WebSocket>();
 export const { rooms, users } = await connectToDb();
 
+/**
+ * Sends updated roomData to all voters in room
+ * @param roomData
+ * @returns void
+ */
+const sendRoomData = (roomData: RoomSchema): void => {
```
Comment (lvl-mattwells) [inline server/controllers/socket.controller.ts:22]: Should this also include sending the room update to the moderator?

Comment (sarashunter) [reply server/controllers/socket.controller.ts:22]: Good catch.  I was thinking that the moderator was also a voter, but I see that might not be the case.  Fixing this.

Comment (sarashunter) [reply server/controllers/socket.controller.ts:22]: Added the logic to update the moderator as well as the voters.

File: server/controllers/socket.controller.ts [hunk 4]
```diff
@@ -53,23 +77,7 @@ async function handleJoin(userId: string, data: JoinEvent): Promise<boolean> {
 		throw new Error(`Updating room failed for room code ${data.roomCode}.`);
```
Comment (sarashunter) [inline server/controllers/socket.controller.ts:77]: We need to decide if we want to throw errors for these or not.   We could move this into the sendRoomData function as well and eliminate duplicated code.

File: server/models/room.model.ts (hunk 2 above)
Comment (lvl-mattwells) [reply server/models/room.model.ts:12]: Makes sense to me!

File: server/controllers/socket.controller.ts (hunk 4 above)
Comment (lvl-mattwells) [reply server/controllers/socket.controller.ts:77]: I think moving it in the sendRoomData function is fine. Less dupe code the better.

Comment (lvl-mattwells) [reply server/controllers/socket.controller.ts]: Seeing some type errors with this change.
image

Comment (sarashunter) [reply server/controllers/socket.controller.ts]: fixed.  thanks.

**Review by lvl-mattwells — APPROVED** [review]
LGTM! A difficult one to test. Will be able to verify everything for sure as part of #30

**Review by misterbastean — APPROVED** [review]
Great job. I especially like that you extracted the sendRoomData method, as well as using an emum for confidence, to make it easier to use without looking up the number values each time.

**Pull request merged**